src/adapter/src/catalog/storage.rs

```diff
@@ -234,6 +261,17 @@ async fn migrate(
                     database_id: Some(MATERIALIZE_DATABASE_ID),
                     name: "public".into(),
                     owner_id: Some(MZ_SYSTEM_ROLE_ID),
+                    privileges: Some(vec![
+                        MzAclItem {
+                            grantee: RoleId::Public,
+                            grantor: MZ_SYSTEM_ROLE_ID,
+                            acl_mode: AclMode::USAGE.bitor(AclMode::CREATE),
```

Comment:
Postgres 15 had a notable release note to

> Remove PUBLIC creation permission on the [public schema](https://www.postgresql.org/docs/15/ddl-schemas.html#DDL-SCHEMAS-PUBLIC)

Should we follow suit? If not, it should probably have a pretty strong reason. Cockroach has been getting requests to do the same as pg15, for example.

Reply:
@mjibson Any thoughts on this? I keep flip-flopping.

On the one hand it's better to default to higher security and let users opt into lower security by modifying the privileges. On the other hand it might be annoying if the first thing every customer has to do with a new Materialize deployment is modify the privileges on the default cluster, DB, and schema.

Reply:
I think we should follow the recommended [usage patterns](https://www.postgresql.org/docs/15/ddl-schemas.html#DDL-SCHEMAS-PATTERNS):

> first ensure that no schemas have public CREATE privileges
> for every user needing to create non-temporary objects, create a schema with the same name as that user

This would also require us to:

- implement `$user` in schema_path, which is probably easy
- tell customers that they have to explicitly create schemas for non-superusers or explicitly GRANT access to existing schemas

I kinda think this is the point of RBAC? Non-superusers can't create anything until explicitly GRANTd access to an existing thing, or given their own place to make things in (`CREATE SCHEMA alice AUTHORIZATION alice`; although I'm not sure we support that `AUTHORIZATION` keyword yet?).

Reply:
This is a good point, I've removed the public create privilege from the "materialize" database, "public" schema, and "default" cluster.

Reply:
> although I'm not sure we support that AUTHORIZATION keyword yet?

We do not, and it wasn't planned as part of this initial RBAC project.